File: Problem 12/ans.py
def primelist(x):
    l=[1]
    for i in range(2,x+1):
        if not x%i:
            l.append(i)
    return len(l)
i=1
diff=1
while True:
    diff+=1
    if primelist(i)>500:
        break
    i+=diff
print(i,"has over 500 divisors")

File: Problem 12/ans.c
#include<stdio.h>

int main()
{
    long int num=1,diff=1,cnt=1,st;
    while (cnt<=500)
    {
        diff++;
        cnt=1;
        for(st=2;st<=num;st++)
            if (!(num%st))
            cnt++;
        if (cnt>500)
            break;
        num+=diff;
    }
    printf("%ld has over 500 divisors",num);
    return 0;
}
